From 4eb41221d26dab18e9a59a756ef49649af23df43 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Kacper=20Michaj=C5=82ow?= <kasper93@gmail.com>
Date: Mon, 2 Dec 2024 17:10:00 +0100
Subject: [PATCH] meson: add glslang lib for 15.0.0 linking

Glslang 15.0.0 moved some code around, add also linking to glslang,
while this is not needed for older versions, it will still work.

(cherry picked from commit 056b852018db04aa2ebc0982e27713afcea8106b)
---
 src/glsl/meson.build | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/glsl/meson.build b/src/glsl/meson.build
index 5cebfb89..4f68da11 100644
--- a/src/glsl/meson.build
+++ b/src/glsl/meson.build
@@ -39,6 +39,10 @@ elif not glslang_req.disabled()
 
     glslang_deps += spirv
 
+    # Glslang 15.0.0 moved some code around, add also linking to glslang, while
+    # this is not needed for older versions, it will still work.
+    glslang_deps += cxx.find_library('glslang', required: not cxx.find_library('glslang-default-resource-limits', required: false).found(), static: static)
+
     if static
       glslang_deps += [
         # Always required for static linking
